modules/Bio/EnsEMBL/Compara/RunnableDB/SpeciesTree/PermuteMatrix.pm

```diff
@@ -154,9 +166,10 @@ sub _add_outgroup {
 
 	my ( $closest_gdb_id, $min_distance) = (undef, 100);
 	foreach my $full_key ( $full_matrix->members ) {
+    print "FULL_KEY: $full_key\n";
 		next if ( defined $self->param('blacklisted_genome_db_ids') && grep { $full_key eq $_ } @{$self->param('blacklisted_genome_db_ids')} ); # skip any that are on the naughty list
 		next if (grep { $_ eq $full_key } @sub_gdb_ids); # skip any that exist in the submatrix
-
+    next unless defined $full_matrix->distance($rep_gdb_id, $full_key);
```

Comment:
Not sure about this line... Is it needed? What happens without it? Should it `die` if there's no distance between some members of the submatrix instead?

Reply:
I will check

Reply:
It was not needed, I have removed it